cms/static/js/factories/import.js

```diff
@@ -1,30 +1,49 @@
 define([
-    'js/views/import', 'jquery', 'gettext', 'jquery.fileupload', 'jquery.cookie'
-], function(Import, $, gettext) {
+    'domReady', 'js/views/import', 'jquery', 'gettext', 'jquery.fileupload', 'jquery.cookie'
+], function(domReady, Import, $, gettext) {
+
     'use strict';
+
     return function (feedbackUrl, library) {
         var dbError;
+
         if (library) {
             dbError = gettext('There was an error while importing the new library to our database.');
         } else {
             dbError = gettext('There was an error while importing the new course to our database.');
         }
+
         var bar = $('.progress-bar'),
             fill = $('.progress-fill'),
             submitBtn = $('.submit-button'),
-            chooseBtn = $('.choose-file-button'),
+            chooseBtn = $('.view-import .choose-file-button'),
             defaults = [
                 gettext('There was an error during the upload process.') + '\n',
                 gettext('There was an error while unpacking the file.') + '\n',
                 gettext('There was an error while verifying the file you submitted.') + '\n',
                 dbError + '\n'
             ],
-            // Display the status of last file upload on page load
-            lastFileUpload = $.cookie('lastfileupload'),
+            previousImport = Import.storedImport(),
             file;
 
-        if (lastFileUpload){
-            Import.getAndStartUploadFeedback(feedbackUrl.replace('fillerName', lastFileUpload), lastFileUpload);
+        Import.callbacks.complete = function () {
```

Comment:
@AkA84 when I see callback I usually ask myself: "Can it be done using Deferred/Promises"? In this case, it looks like it can: basically `Import.start` and `Import.resume` would return deferred, which is than resolved when success/error event occurs. Could you please try using deferred here, as it usually yield cleaner and more maintainable code.